cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -315,6 +317,13 @@ private void maybeAddLeaderChangeTasks(Collection<ExecutionProposal> proposals,
                                                                 Set<TopicPartition> inProgressPartitions) {
     LOG.trace("Getting inter-broker replica movement tasks for brokers with concurrency {}", readyBrokers);
     List<ExecutionTask> executableReplicaMovements = new ArrayList<>();
+
+    // Kafka doesn't allow new replica movements if there is an ongoing rebalance
+    if (!inProgressPartitions.isEmpty()) {
```

Comment:
Would this alliviate the issue described in https://github.com/linkedin/cruise-control/issues/1167#issuecomment-620314334 ? @efeg can you please comment?

Reply:
I do not think this check should be added.

The way CC submit tasks is in batch. we select a set of tasks based on the concurrency requirement and submit them in batch, later we check the finish status and select more tasks to current ongoing tasks(as long as we do not violate concurrency requirement) and submit in second batch.

Reply:
From my understanding, Kafka cannot start a new rebalance while a current one is in progress. So we don't need to create a second batch, if the first batch isn't completed yet.

But I'll remove this line, since it isn't related to the referenced issue -- and isn't important for the bugfix.

Reply:
Kafka can accept new task while there is ongoing task and that CC use that to ensure the task are always executed in desired concurrency.

Reply:
@matthinograb
Just wanted to chime in to give a little more context on @kidkun 's explanation. 
Starting a new rebalance while a current one is in progress requires the workaround patch described in https://github.com/linkedin/cruise-control/issues/1167#issuecomment-620314334 -- i.e. it is indeed not supported in Apache Kafka out of the box.

Hope it helps!